model.js

```diff
@@ -17,80 +17,114 @@ exports.fetchArticleById = (article_id) => {
     });
 };
 
-exports.fetchArticles = (sort_by = "created_at", order = "DESC") => {
+exports.fetchArticles = (sort_by = "created_at", order = "DESC", topic) => {
   let sqlQueryString = `SELECT articles.*,
     COUNT(comments.article_id)::INT AS comment_count
     FROM comments
     RIGHT JOIN articles
-    ON articles.article_id = comments.article_id
-    GROUP BY articles.article_id `;
+    ON articles.article_id = comments.article_id `;
 
-  sqlQueryString += `ORDER BY ${sort_by} ${order}`;
+  let topicValue = [];
 
-  return db.query(sqlQueryString).then(({ rows }) => {
+  if (topic) {
+    sqlQueryString += `WHERE topic LIKE $1 `;
+    topicValue.push(topic);
+  }
+
+  sqlQueryString += `GROUP BY articles.article_id ORDER BY ${sort_by} ${order}`;
+
+  return db.query(sqlQueryString, topicValue).then(({ rows }) => {
+    if (rows.length === 0) {
+      return Promise.reject({ status: 404, msg: "Non-existent Topic" });
+    }
     return rows;
   });
 };
 
-exports.fetchCommentsByArticleID = (article_id,sort_by = "created_at", order = "DESC") =>{
-    let sqlQueryString = `SELECT comments.* 
+exports.validateQuery = (queries) => {
```

Comment:
interesting way of doing it! I like that it's "future-proof" and you could potentially add new columns to the db and you won't need to change your query validation at all, really nicely done